src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/controller/ServiceClientHelper.java

```diff
@@ -104,4 +105,27 @@ public BadRequestException wrapInBadRequestException(
         return new BadRequestException(INVALID_SERVICE_CLIENT_ID + e.getServiceClientIdentifier(),
                 new ErrorDeviation(ERROR_INVALID_SERVICE_CLIENT_ID));
     }
+
+    /**
+     * Sort a list of ServiceClient objects using member name / group description as the primary sort key, and client id
+     * as the secondary sort key.
+     * @param clients
+     */
+    public void sortServiceClientsList(List<ServiceClient> clients) {
+        clients.sort((c1, c2) -> {
+                if (c1.getName() == null && c2.getName() == null) {
+                    return c1.getId().compareToIgnoreCase(c2.getId());
```

Comment:
I would add some tests to verify just the ordering (without any controllers, services, etc) and the logic for different null / not null cases etc. 

In fact, I would usually start with the test and then write the implementation, I find that the easiest way to reach correct result in details like "should null here be +1 or -1...?". Especially since the test should be easy to implement (building a few `ServiceClient` objects with wanted name & id fields).

Reply:
Added tests for both comparators that verify the comparison logic.